department-of-veterans-affairs/abd-vro
# Standup mock-bgs-api via docker compose

## What was the problem?
Updates mock-bgs-api to allow calling mocked `createNote` endpoint.

Associated tickets or Slack threads:
- #2190

Blocks:
- #2191 

Replaces:
- #1810 

## How does this fix it?[^1]

This allows running mock-bgs-api via docker compose, and hitting the createNote endpoint and returning the mocked response.

## How to test this PR
- run the mock service by:
```
COMPOSE_PROFILES=bgs ./gradlew -p mocks docker
COMPOSE_PROFILES=bgs ./gradlew -p mocks :dockerComposeUp
```
- Follow instructions in readme to send a test message to the createNote endpoint
- Verify curl response is 200 with mocked response
- You can also verify logs in the docker container by navigating to the file `/usr/local/tomcat/logs/localhost_access_log.<date>.txt`

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
